Stream of consciousness rough draft (#1543)
haacked/haacked.com: _posts/2022/2022-07-25-lessons-from-the-pivot.md

---
title: "Lessons From a Startup Pivot"
description: "Depending on SAAS, PAAS, and IAAS is great, but it can hurt when you should have turned left at Albuquerque if you haven't insulated yourself from some of those choices."
tags: [aspnetcore, azure, cloud]
excerpt_image: https://user-images.githubusercontent.com/19977/180076597-6fbdb672-6539-43fd-b10d-1a1e667c8d8b.png
---

Building a startup is easy. You file some paperwork and bam! You're a startup!

> Building a start-up that's sustainable and can pay everyone a nice salary, on the other hand, is very tough. Today, as my company, [A Serious Business, Inc.](https://www.aseriousbusiness.com/) launches [Abbot on Product Hunt](https://www.producthunt.com/posts/abbot), I thought I'd reflect on some of the lessons learned in the past couple years.

Reviewer: ```suggestion
Building a start-up that's sustainable and can pay everyone a nice salary, on the other hand, is very tough. We are launching [Abbot on Product Hunt](https://www.producthunt.com/posts/abbot) today, and preparing last week made me reflect on some of the lessons we've learned since starting our business.
```
Feels like this could be shortened given the next line describes what Abbot is.

> [Abbot](https://ab.bot/) is a Slack App (and bot) that helps teams keep track of customer conversations in Slack.

Reviewer: ```suggestion
[Abbot](https://ab.bot/) is a Slack App (and bot) that helps teams scale their customer conversations in Slack.
```
Abbot does more than keep track of conversations :)

Some of you might be thinking, "Wait! I thought Abbot was [ChatOps as a service](https://news.ycombinator.com/item?id=27974077)?" Indeed, that was our original goal, but we found it tough to to sell. Conversations often would go like this:

> them: What problem do you solve?
> us: What problems do you have? Abbot can help with them all!
> them: *snoring*

Yeah, not too effective as a sales pitch.

> ## Lesson: Articulate a specific problem your product solves

Reviewer: I feel like "build something people want" is well trodden advice. This section could probably be cut if you're looking to trim length.

I know this isn't new or groundbreaking advice. In fact, it's one of the first things you hear when people give advice about building a company. But it's easy to fall into the trap of thinking that *your* product is going to be the exception.

It's not that Abbot wasn't interesting, useful, and wonderful. A lot of folks told us, "That's very cool! There's so much I could do with that! I hope to find some time to experiment with it." And there's the rub. People are busy and they don't have time to solve their own problems for them. They want your product to solve their problems. And it needs to be a concrete specific problem that matters, not something hypothetical. This leads us right to the next lesson.

## Lesson: Start with selling a Product not a Platform

> Abbot started off as an automation platform. With Abbot, we could quickly write and host new capabilities of the bot from within Abbot. We used it internally to solve whole classes of problems and run experiments. Platforms give people the tools to solve whole classes of problems, but they don't, on their own, solve any specific problems. And this is why platforms are hard to sell. For example, Amazon didn't begin with selling their Amazon Web Services (AWS) platform. They started with an online bookstore. A concrete product. And over time, they reached the point where they could sell the underlying platform (AWS) the bookstore was built on.

Reviewer: ```suggestion
Abbot started off as an automation platform. With Abbot, we could quickly write and host new capabilities of the bot from within Abbot. We used it internally to solve whole classes of problems and run experiments. Platforms give people the tools to solve whole classes of problems, but they don't solve any specific problems on their own. And this is why platforms are hard to sell. For example, Amazon didn't begin with selling their Amazon Web Services (AWS) platform. They started with an online bookstore. A concrete product. And over time, they reached the point where they could sell the underlying platform (AWS) the bookstore was built on.
```

Before I get into the problems we ran into, it's important to understand some context, lest you think we're running a clown shoes operation. When we joined the YCombinator startup program, one of the key lessons drilled into us is that Product Market Fit is our top priority. And the reasoning is simple, if you don't achieve product market fit, then you don't have a business and your company runs out of money and dies. And it's hard to build great software with a dead company. So while good engineering and infrastructure is important, if you spend too much time on that, but don't achieve product market fit, then all that exquisite infrastructure goes to waste.

At the same time, if you wait too long to address your infrastructure needs, your product may fail under the weight of all that juicy product market fit. It's definitely a delicate balancing act and one that you have to feel out for yourself. I hope that sharing some of our failures can inpspire some ideas on how to strike that balance for your own product and company. That balance is going to be different for every product and company.

## Lesson: All Services Are Leaky

> Around the time dinosaurs roamed the wold hosting their own server racks, I was a fresh-faced college graduate starting my first job at a custom development shop. We had a full server rack in the closet where we hosted all our clients web sites. This put us in control of our own hardware, software, and destiny. Which as it turns out, isn't always a good thing. I remember the time my boss wanted to demonstrate how our new hard-drive array was hot-swappable by pulling out a drive. He pulled out the drive and everything continued to work flawlessly. Just kidding. It all came crashing down like a soccer player who feels a gentle tug on the shirt in the penalty box (if the sport metaphor doesn't work for you, just trust me on this one).

Reviewer: ```suggestion
Around the time dinosaurs roamed the world hosting their own server racks, I was a fresh-faced college graduate starting my first job at a custom development shop. We had a full server rack in the closet where we hosted all our clients' web sites. This put us in control of our own hardware, software, and destiny. Which as it turns out, isn't always a good thing. I remember the time my boss wanted to demonstrate how our new hard-drive array was hot-swappable by pulling out a drive. He pulled out the drive and everything continued to work flawlessly. Just kidding. It all came crashing down like a soccer player who feels a gentle tug on the shirt in the penalty box (if the sport metaphor doesn't work for you, just trust me on this one).
```

At the time, this was the way you built web software. You hosted everything.

Fast forward a bunch of years, and the pendulum swung to the other side. I didn't want to host a damn thing. And with tools like Heroku, Azure Web Sites, AWS, Google Cloud, etc. I really didn't need to. Much of my later career at Microsoft and GitHub was spent building developer tools and client applications. Occasionally, I'd build a website on the side for fun. I'd just deploy it to a cloud provider, take advantage of all their services, and not worry about it too much.

> And this approach worked well! It worked so well, when I started a company with my colleague, Paul, we were able to [get a site up](https://ab.bot/) and running in no time. And this all worked great, everyone was happy, and it all just worked out...Until we needed to pivot.

Reviewer: Maybe link to the DNR episode about the pivot?

Reviewer: https://www.dotnetrocks.com/default.aspx?ShowNum=1791
